test/Sentry.Tests/Internals/Http/CachingTransportTests.cs

```diff
@@ -0,0 +1,227 @@
+
+            var innerTransport = Substitute.For<ITransport>();
+            var isFailing = true;
+
+            innerTransport
+                .SendEnvelopeAsync(Arg.Any<Envelope>(), Arg.Any<CancellationToken>())
+                .Returns(_ =>
+                    isFailing
+                        ? new ValueTask(Task.FromException(new IOException()))
+                        : new ValueTask()
+                );
+
+            using var transport = new CachingTransport(innerTransport, options);
+
+            // Can't really reliably test this with a worker
+            await transport.StopWorkerAsync();
+
+            // Act
+            for (var i = 0; i < 3; i++)
+            {
+                using var envelope = Envelope.FromEvent(new SentryEvent());
+                await transport.SendEnvelopeAsync(envelope);
+            }
+
+            await transport.FlushAsync();
+
+            // (transport stops failing)
+            innerTransport.ClearReceivedCalls();
+            isFailing = false;
+            await transport.FlushAsync();
+
+            // Assert
+            // (3 envelope retried)
+            _ = innerTransport.Received(3).SendEnvelopeAsync(Arg.Any<Envelope>(), Arg.Any<CancellationToken>());
+        }
+
+        [Fact(Timeout = 10000)]
+        public async Task DoesNotRetryOnNonTransientExceptions()
+        {
+            // Arrange
+            var options = new SentryOptions {CacheDirectoryPath = CacheDirectoryPath};
+
+            var innerTransport = Substitute.For<ITransport>();
+            var isFailing = true;
+
+            innerTransport
+                .SendEnvelopeAsync(Arg.Any<Envelope>(), Arg.Any<CancellationToken>())
+                .Returns(_ =>
+                    isFailing
+                        ? new ValueTask(Task.FromException(new HttpRequestException()))
+                        : new ValueTask()
+                );
+
+            using var transport = new CachingTransport(innerTransport, options);
+
+            // Can't really reliably test this with a worker
+            await transport.StopWorkerAsync();
+
+            // Act
+            for (var i = 0; i < 3; i++)
+            {
+                using var envelope = Envelope.FromEvent(new SentryEvent());
+                await transport.SendEnvelopeAsync(envelope);
+            }
+
+            await transport.FlushAsync();
+
+            // (transport stops failing)
+            innerTransport.ClearReceivedCalls();
+            isFailing = false;
+            await transport.FlushAsync();
+
+            // Assert
+            // (0 envelopes retried)
+            _ = innerTransport.Received(0).SendEnvelopeAsync(Arg.Any<Envelope>(), Arg.Any<CancellationToken>());
+        }
+
+        public void Dispose()
+        {
+            if (Directory.Exists(CacheDirectoryPath))
+            {
+                Directory.Delete(CacheDirectoryPath, true);
+            }
+        }
```

Comment:
It just noops if there's no folder to delete

```suggestion
            Directory.Delete(CacheDirectoryPath, true);
        }
```

Reply:
No it doesn't 😄